### PR TITLE
add intl to korean and arabic + improved packaged dependency checks

### PR DESCRIPTION
**Pull request by pklpriv**
This PR adds sets new tokenizers for Korean and Arabic.
These require special installation instruction, so added enhanced mechanism in PackageRequirementsMixin

class PackageRequirementsMixin(Artifact):
    """Base class used to automatically check for the existence of required python dependencies for an artifact (e.g. Operator or Metric).

    The _requirement list is either a list of required packages
    (e.g. ["torch","sentence_transformers"]) or a dictionary between required packages
    and detailed installation instructions on how how to install each package.
    (e.g. {"torch" : "Install Torch using `pip install torch`", "sentence_transformers" : Install Sentence Transformers using `pip install sentence-transformers`})
    Note that the package names should be specified as they are used in the python import statement for the package.
    """

### CODE REVIEW

**Review by yoavkatz — CHANGES_REQUESTED** [review]
Please add checks in the prepare to the new languages. Similar to this check in English.

predictions = ["hello there general kenobi", "on our way to ankh morpork"]
references = [
    ["hello there general kenobi", "hello there !"],
    ["goodbye ankh morpork", "ankh morpork"],
]
instance_targets = [
    {
        "counts": [4, 3, 2, 1],
        "totals": [4, 3, 2, 1],
        "precisions": [1.0, 1.0, 1.0, 1.0],
        "bp": 1.0,
        "sys_len": 4,
        "ref_len": 4,
        "sacrebleu": 1.0,
        "score": 1.0,
        "score_name": "sacrebleu",
    },
    {
        "counts": [2, 1, 0, 0],
        "totals": [6, 5, 4, 3],
        "precisions": [0.33, 0.20, 0.12, 0.08],
        "bp": 1.0,
        "sys_len": 6,
        "ref_len": 3,
        "sacrebleu": 0.16,
        "score": 0.16,
        "score_name": "sacrebleu",
    },
]
global_target = {
    "sacrebleu": 0.4,
    "score": 0.4,
    "counts": [6, 4, 2, 1],
    "totals": [10, 8, 6, 4],
    "precisions": [0.60, 0.50, 0.33, 0.25],
    "bp": 1.0,
    "sys_len": 10,
    "ref_len": 7,
    "score_name": "sacrebleu",
    "score_ci_low": 0.11,
    "score_ci_high": 1.0,
    "sacrebleu_ci_low": 0.11,
    "sacrebleu_ci_high": 1.0,
}
outputs = test_metric(
    metric=metric,
    predictions=predictions,
    references=references,
    instance_targets=instance_targets,
    global_target=global_target,
)

**Comment by pklpriv** [comment]
> Please add checks in the prepare to the new languages. Similar to this check in English.
> 
> predictions = ["hello there general kenobi", "on our way to ankh morpork"] references = [ ["hello there general kenobi", "hello there !"], ["goodbye ankh morpork", "ankh morpork"], ] instance_targets = [ { "counts": [4, 3, 2, 1], "totals": [4, 3, 2, 1], "precisions": [1.0, 1.0, 1.0, 1.0], "bp": 1.0, "sys_len": 4, "ref_len": 4, "sacrebleu": 1.0, "score": 1.0, "score_name": "sacrebleu", }, { "counts": [2, 1, 0, 0], "totals": [6, 5, 4, 3], "precisions": [0.33, 0.20, 0.12, 0.08], "bp": 1.0, "sys_len": 6, "ref_len": 3, "sacrebleu": 0.16, "score": 0.16, "score_name": "sacrebleu", }, ] global_target = { "sacrebleu": 0.4, "score": 0.4, "counts": [6, 4, 2, 1], "totals": [10, 8, 6, 4], "precisions": [0.60, 0.50, 0.33, 0.25], "bp": 1.0, "sys_len": 10, "ref_len": 7, "score_name": "sacrebleu", "score_ci_low": 0.11, "score_ci_high": 1.0, "sacrebleu_ci_low": 0.11, "sacrebleu_ci_high": 1.0, } outputs = test_metric( metric=metric, predictions=predictions, references=references, instance_targets=instance_targets, global_target=global_target, )

@yoavkatz How do we actually create those `predictions` and `references` sentences? Is there any rule to pick those from dataset?

**Comment by elronbandel** [comment]
I think they are randomly picked. Best to choose examples the reflect a real data as much as possible. Cc @yoavkatz .

**Comment by yoavkatz** [comment]
Thanks.  Perhaps worth while to add another instance for each case, where the prediction=reference, to see the score is indeed 1.0?  This is done in English and Japanese.

**Comment by pklpriv** [comment]
> Thanks. Perhaps worth while to add another instance for each case, where the prediction=reference, to see the score is indeed 1.0? This is done in English and Japanese.

@yoavkatz Sure, I will add them. For Japanese there is only one as well and it is opposite case - we only have test for pred=ref. I think it is worth to add such for Japanese as well.

**Comment by pklpriv** [comment]
@yoavkatz @elronbandel 

To install 'ko-mecab' , MacOS users may need to install `mecab` with brew with such command:
`brew install mecab`

Only with that I had installation of `sacrebleu[ko]` successful. There could be update of some README. I could create some bash script to run with setup. Such script could check OS of user and install mecab with brew accordingly.

If we go with one of options above or something different - is not my decision to make, therefore I draft that PR until it is resolved

**Comment by yoavkatz** [comment]
> @yoavkatz @elronbandel
> 
> To install 'ko-mecab' , MacOS users may need to install `mecab` with brew with such command: `brew install mecab`
> 
> Only with that I had installation of `sacrebleu[ko]` successful. To inform user about that, we could update README. I could create some bash script to run with setup. Such script could check OS of user and install mecab with brew accordingly.
> 
> If we go with one of options above or something different - is not my decision to make, therefore I draft that PR until it is resolved
 
 To be consistent with other metrics, I think the right way to do this, is that if the metric does not load, print a message with the needed installation  instructions.   Unitxt minimized installation requirements, so we don't want to require such an installation when not required.

**Comment by pklpriv** [comment]
@elronbandel @yoavkatz 

I added check for libraries and additional examples for tests.

**Review by yoavkatz — APPROVED** [review]

File: src/unitxt/metrics.py [hunk 1]
```diff
@@ -3309,3 +3309,30 @@ def compute(
                 best_thr = thr
 
         return {self.main_score: best_acc, "best_thr_max_acc": best_thr}
+
+
+KO_ERROR_MESSAGE = """
+
+Additional dependencies required. To install them, run:
+`pip install "sacrebleu[ko]"`.
+bre
```
Comment (yoavkatz) [inline src/unitxt/metrics.py:3318]:
```suggestion

```

**Review by elronbandel — APPROVED** [review]

**Pull request merged**